src/components/data-panel/preview.tsx

```diff
@@ -0,0 +1,26 @@
+import { useEffect, useRef } from 'react';
+import { ListTable } from '@visactor/vtable';
+
+export const Preview = (props: { data: any[] }) => {
+  const containerRef = useRef<HTMLDivElement>();
+
+  useEffect(() => {
+    if (!containerRef?.current || props.data.length === 0) {
+      return;
+    }
+    const columns = Object.keys(props.data[0]).map(field => ({
+      field: field,
+      title: field,
+      width: 'auto'
+    }));
+
+    const vtable = new ListTable(containerRef.current, {
+      records: props.data,
+      columns,
+      widthMode: 'adaptive'
+    });
+    return () => vtable.release();
+  }, []);
```

Comment:
Does it work correctly without adding props.data to the useEffect deps?

Reply:
done